static/js/server_events_dispatch.js

```diff
@@ -710,6 +712,14 @@ export function dispatch_normal_event(event) {
                     status_text: event.status_text,
                 });
                 activity.redraw_user(event.user_id);
+
+                // Update the status text in compose box placeholder when opened to self.
+                if (
+                    compose_state.get_message_type() === "private" &&
+                    compose_state.private_message_recipient() === page_params.email
```

Comment:
We should do something off user ID, not email. Also, we need to update the status of other users too!  Easy to test with multiple browsers.

I think cleanest would be to just check if `event.user_id` is in `compose_pm_pill.get_user_ids`, and if so, call the redraw function.  (Possibly with an added check for if `compose_pm_pill` has valid data; not sure).

Reply:
Which `redraw` function? I searched but couldn't found any relevant redraw function. Did you meant `compose_actions.update_placeholder_text` only?

I checked, `compose_pm_pill` always has valid data i.e If you open the compose box with some PM recipients and close it, `compose_pm_pill.get_user_ids` will return an empty array.

Reply:
Yeah that's what I meant.